sklearn/random_projection.py

```diff
@@ -536,6 +542,11 @@ def _make_random_matrix(self, n_components, n_features):
             n_components, n_features, random_state=random_state
         )
 
+    def _more_tags(self):
+        return {
+            "preserves_dtype": [np.float64, np.float32],
+        }
```

Comment:
I think we can define `_more_tags` once in the base class `BaseRandomProjection` and remove the two definitions from `GaussianRandomProjection` and `SparseRandomProjection`.

```suggestion
```